app/helpers/application_helper.rb

```diff
@@ -3,33 +3,30 @@ module ApplicationHelper
   include HomeHelper
 
   # Add autogenerated html data-id attribute if not present (prefixed with "aid_")
-  def link_to(*args, &block)
-    if block_given?
-      super(*args)
-    else
-      options = args[1] || {}
-      html_options = args[2] || {}
-      unless html_options.has_key?(:'data-id') || (options.is_a?(String) && options.starts_with?("mailto:"))
-        begin
-          path = URI.split(url_for(options) || html_options['href'])[5].split(/\//).select {|x| !x.empty?}
-          if !path.empty?
-            max = (path.size <= 3) ? path.size : 3
-            id = path.last(max).join('_')
-          else
-            id = 'not_defined'
-          end
-        rescue => e
-          Foreman::Logging.exception("Failed generating link using #{args.inspect}", e)
-          id = 'not_parseable'
-        end
-        html_options[:'data-id'] = "aid_#{id}"
+  def link_to(*args)
+    return super(*args) if block_given?
+
+    options = args[1] || {}
+    html_options = args[2] || {}
+    unless html_options.has_key?(:'data-id') || (options.is_a?(String) && options =~ /^(mailto|javascript):/)
```

Comment:
Why not leave it as `starts_with?`? it's much more readable than a regex

Reply:
It doesn't make it more readable when you start adding many options, but clatter the boolean validation here.

Imagine something like this:

`unless html_options.has_key?(:'data-id') || (options.is_a?(String) && options.start_with?('mailto:', 'javascript:', 'tel:', 'color:', 'date:', 'foo:'))`

It takes you few seconds to read it and then realize that each of it is an option that we are are looking for to see if the string start with, but actually the regex makes it simpler, because if we read it from left to right, we see the carrot sign and then a group of words that we are looking for at the beginning, and only then a colon sign.

So I think that the more options you have, `start_with?` because a bit less readable, and makes you think twice or more of the meaning, at least for me :/